### DISCUSSION TITLE
Initialize Damage from Initial Strain

### DISCUSSION THREAD

**Discussion by SyazwanSamad** [OpenRadioss/OpenRadioss#529] [Ideas]
If not mistaken, starting from RADIOSS 14.0, we can import damage value from .sta file. 
What if we don't have the damage value from the .sta file? (especially if we run the forming simulation using another software)
I'm not sure whether this idea has been implemented or not, but I believe that we can implement damage history due to manufacturing history from the initial strain tensor. I implement the following code in /FAIL/** subroutine to calculate the damage at T=0 for shell elements by assuming the strain path is linear. The code is tested using tensile simulation at very low plastic strain (<10%). The first simulation is a normal run while the other is initialized using .sta output from the normal run. The difference of damage is 0.02, which I think it is acceptable.

Note: I'm using failure surface (plastic strain vs stress triaxiality vs lode param) is used instead of failure curve, that is the reason why lode angle is added in equation below.

C---------INITIALIZE DAMAGE DUE TO MANUF. HISTORY-------------
C---------ASSUME LINEAR STRAIN PATH---------------------------
       IF (TIME == ZERO) THEN
        DO I = 1,NEL
         IF (PLA(I) /= ZERO) THEN
           E12= HALF*EPSXY(I)
           S1 = HALF*(EPSXX(I) + EPSYY(I))
           S2 = HALF*(EPSXX(I) - EPSYY(I))
           Q  = SQRT(S2**2 + E12**2)
           EMAJ(I) = S1 + Q
           EMIN(I) = S1 - Q
           IF (EMIN(I) >= EMAJ(I)) THEN
             SS      = EMIN(I)
             EMIN(I) = EMAJ(I)
             EMAJ(I) = SS
           ENDIF
           BETA = EMIN(I)/EMAJ(I)
C -------ST = Stress Triaxiality-----------------
           ST = (1+BETA)/SQRT(THREE)/SQRT(1+BETA+BETA**2)
C -------LODEP = Lode Parameter-----------------
C           LODEP = ONE-(TWO/PI*ACOS(-TWENTY7*HALF*ST*(ST**2-THIRD)))
           LODEP = ASIN((-TWENTY7*HALF*ST)*(ST**2-THIRD))*2/PI
           EPSFAIL = C1+C2*ST+C3*LODEP+C4*ST**2+C5*LODEP**2+C6*ST*LODEP
           UVAR(I,1) = PLA(I)/EPSFAIL
           DFMAX(I) = PLA(I)/MAX(EPSFAIL,EM6)
           DFMAX(I) = MIN(ONE,DFMAX(I))
         ENDIF
        ENDDO
       ENDIF

**Comment by pmondalek** [comment]
Hello,
Thank you for your comment.
Yes you can use the /STATE/SHELL/FAIL to get the failure variables.
I am not sure i understood what you are using in your normal run, is it the code you wrote here with what you have in your manufactoring hsitory?
In your second run what are you using with /INISHE?
Regards

**Comment by SyazwanSamad** [comment]
Hello.
The first run is just a normal simulation with /STATE output. The codes that I posted here converts the /INISHE/STRA_F input to DAMAGE value (in the second run). Imagine the first run is just like using other solvers such as LS-Dyna onestep, it is not possible to get the DAMAGE value calculated by LS-Dyna. Scripts are required to convert the LS_Dyna output to RADIOSS format. 

That's what triggers me to ask about this idea, instead of doing a manual conversion, why not implement it in solver level?
Figure below is a proof of concept of converting strain value to DAMAGE value. I haven't validate it to large scale model yet. I intend to do so later.
![image](https://user-images.githubusercontent.com/113438036/192568400-39961836-51a9-4998-bc1b-485a28adf74e.png)
 
Regards

**Comment by pmondalek** [comment]
Hello
this might be possible for constant loading path, but not sure it stays accurate to compute damage this way when you are not linear and when the loading path varies and you need the history. 
Besides it is also relative to every failure model.
It is an interesting idea, to be investigated with other loading cases.
Thanks

**Comment by SyazwanSamad** [comment]
Hello, I definitely agree with your statement regarding the loading/strain path. However, most of vehicle manufacturer is using onestep forming simulation to incorporate manufacturing histories into their complete vehicle model, I'm not sure whether we can get a complete loading path out of this type of simulation (can we?). 
If they are using incremental forming simulation, definitely this method is not accurate/relevant.
Regards

**Reply by MarianBulla** [reply]
Hello, yes, it is definitely an interesting idea and worth to have a deepler look at it.
For sure, it depends on the loading path and we should investigate the damage accumulation depending on the state of stress. 
It can increase the quality of results, expecially when using one-step forming solutions.
Anyway, this is a very good idea.
Thank you very much for bringing this in!

**Reply by SyazwanSamad** [reply]
Hi Marian, thank you for your feedback. I encountered a lot of this issue when I was in automotive engineering, hence, I came up with this idea. However, I don't have enough time and resources to validate this idea. When the Radioss solver has become opensource, I took the opportunity to check this out.

**Comment by MaciekWronski** [comment]
Hi,
interesting approach indeed, but difficult to generalize because of the impossibility to estimate errors. Your first test is special case which clearly minimises these errors. First, assumption that strain and stress tesnors are colinear (true in linear case or very low plasticity), then of course, constant Lode angle during load (mostly true in simple tension sample). Otherwise, it may largely diverge. 
That said, when no other method of retrieving damage is available, it might be always better than nothing. Le'ts just make sure that we understand it's limitations. I am curious to see  results of more complex model, like stamping for example.
Thanks for interesting topic and discussion !
Maciek

**Reply by SyazwanSamad** [reply]
I planned to test the concept against one step forming simulation, and afterwards, against incremental forming simulation. Thank you for your comment!

**Reply by MarianBulla** [reply]
Yes, it is definitely a worth development to be continued.
Much appreciated !
